gemd/demo/cake.py

```diff
@@ -639,12 +773,23 @@ def ingredient_kwargs(material):
     )
 
     # Crawl tree and annotate with uids; only add ids if there's nothing there
-    recursive_foreach(cake, lambda obj: obj.uids or obj.add_uid(DEMO_SCOPE, obj.name))
+    seen = set()
+
+    def name_fuzz(obj):
+        # Add fuzz to name in ID as necessary
+        nonlocal seen
```

Comment:
This non-local pattern looks to work, but I think the slightly more pythonic way would be to add seen to the argument list of name_fuzz (and maybe a leading underscore).

Reply:
Migrated to using a closure & generator.